File: services/user-service/src/main/java/com/visor/school/userservice/dto/RegisterRequest.java
package com.visor.school.userservice.dto;

import java.util.Set;

import com.visor.school.userservice.model.UserRole;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;

public record RegisterRequest(
    @Email @NotBlank String email,
    @NotBlank String firstName,
    @NotBlank String lastName,
    Set<UserRole> roles,
    @NotBlank String password,
    String phoneNumber
) {}

File: services/user-service/src/main/kotlin/com/visor/school/userservice/dto/RegisterRequest.kt
package com.visor.school.userservice.dto

import com.visor.school.userservice.model.UserRole
import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank

// Request DTOs
data class RegisterRequest(
    @field:Email
    @field:NotBlank
    val email: String,

    @field:NotBlank
    val firstName: String,

    @field:NotBlank
    val lastName: String,

    val role: UserRole,

    @field:NotBlank
    val password: String,

    val phoneNumber: String? = null
)